File: internal/cli/user/usage.go
package user

import (
	"fmt"
	"strings"

	"github.com/spf13/cobra"
)

const usageText = `lin user — Look up Linear workspace users

SUBCOMMANDS:
  user search <text>         Search users by name, email, or display name
  user list [--team <team>]  List users (optionally filtered by team)
  user me                    Current authenticated user + organization

OPTIONS (search):
  --limit <n>               Limit results
  --cursor <token>          Pagination cursor for next page

OPTIONS (list):
  --team <team>             Filter by team key or name (e.g., "ENG")
  --limit <n>               Limit results
  --cursor <token>          Pagination cursor for next page

OUTPUT FIELDS:
  search → id, name, email, displayName
  list   → id, name, email, displayName
  me     → id, name, email, displayName, organization (id, name)

NOTES:
  User IDs are UUIDs. Use "user me" to get the current user's ID for
  operations like --assignee filtering.
  When --team is specified, returns only members of that team.
  Search matches against name, email, and displayName (case-insensitive).`

func registerUsage(user *cobra.Command) {
	cmd := &cobra.Command{
		Use:   "usage",
		Short: "Print detailed user command documentation (LLM-optimized)",
		Args:  cobra.NoArgs,
		Run: func(_ *cobra.Command, _ []string) {
			fmt.Println(strings.TrimSpace(usageText))
		},
	}
	user.AddCommand(cmd)
}

File: bun/src/cli/label/usage.ts
import type { Command } from "commander";

const USAGE_TEXT = `lin label — List Linear issue labels

SUBCOMMANDS:
  label list [--team <team>]  List labels (optionally filtered by team)

OPTIONS:
  --team <team>             Filter by team key or name (e.g., "ENG")
  --limit <n>               Limit results
  --cursor <token>          Pagination cursor for next page

OUTPUT FIELDS:
  list → id, name, color

NOTES:
  Without --team, returns all workspace labels.
  With --team, returns only labels scoped to that team.
  Label names are used as values for --label filters and "issue update labels".
`;

export function registerUsage(label: Command): void {
  label
    .command("usage")
    .description("Print detailed label command documentation (LLM-optimized)")
    .action(() => {
      console.log(USAGE_TEXT.trim());
    });
}
